src/OpenTracing/ITracer.cs

```diff
@@ -0,0 +1,31 @@
+﻿using OpenTracing.Propagation;
+
+namespace OpenTracing
+{
+    public interface ITracer
+    {
+        /// <summary>
+        /// Create, start, and return a new Span.
+        /// </summary>
+        /// <param name="operationName">The operation name of the Span.</param>
+        /// <param name="options">Gives control over the start timestamp and references to other Spans.</param>
+        ISpan StartSpan(string operationName, StartSpanOptions options = null);
+
+        /// <summary>
+        /// This takes the <paramref name="spanContext"/> and injects it for propagation within <paramref name="carrier"/>. 
+        /// The actual type of <paramref name="carrier"/> depends on the value of <paramref name="format"/>.
+        /// </summary>
+        /// <param name="spanContext">The <see cref="ISpanContext"/> instance to inject into the carrier.</param>
+        /// <param name="format">The format which should be used.</param>
+        /// <param name="carrier">See the documentation for the chosen <paramref name="format"/> for a description of the carrier object.</param>
+        void Inject(ISpanContext spanContext, string format, IInjectCarrier carrier);
```

Comment:
(would be interested to read a response to @yurishkuro's comment, here or where he linked to)